src/uffs/uffs_fs.c

```diff
@@ -557,6 +557,71 @@ URET uffs_OpenObjectEx(uffs_Object *obj, uffs_Device *dev,
 	return (obj->err == UENOERR ? U_SUCC : U_FAIL);
 }
 
+URET uffs_OpenObjectBySerial(uffs_Object *obj, uffs_Device *dev, int serial, int oflag)
+{
+		obj->err = UENOERR;
+	obj->open_succ = U_FALSE;
+
+	if (dev == NULL) {
+		obj->err = UEINVAL;
+		goto ext;
+	}
+
+	if ((oflag & (UO_WRONLY | UO_RDWR)) == (UO_WRONLY | UO_RDWR)) {
+		/* UO_WRONLY and UO_RDWR can't appear together */
+		uffs_Perror(UFFS_MSG_NOISY,
+					"UO_WRONLY and UO_RDWR can't appear together");
+		obj->err = UEINVAL;
+		goto ext;
+	}
+
+	obj->dev = dev;
+	obj->type = (oflag & UO_DIR ? UFFS_TYPE_DIR : UFFS_TYPE_FILE);
+
+	uffs_ObjectDevLock(obj);
+
+	if (obj->type == UFFS_TYPE_DIR) {
```

Comment:
In `uffs_OpenObejctEx`, when you try to open root directory (with serial = `ROOT_DIR_SERIAL`) then `obj->node` is not propagated. See line 499 in this file

Reply:
done